src/JsonSchema/Constraints/Factory.php

```diff
@@ -73,9 +73,43 @@ public function __construct(
         UriRetrieverInterface $uriRetriever = null,
         $checkMode = Constraint::CHECK_MODE_NORMAL
     ) {
+        // set provided config options
+        $this->setConfig($checkMode);
+
         $this->uriRetriever = $uriRetriever ?: new UriRetriever;
         $this->schemaStorage = $schemaStorage ?: new SchemaStorage($this->uriRetriever);
-        $this->checkMode = $checkMode;
+    }
+
+    /**
+     * Set config values
+     *
+     * @param int $checkMode Set checkMode options
+     * @param bool $enable Whether to enable or disable the provided options. If null, overwrites checkMode.
+     */
+    public function setConfig($checkMode = Constraint::CHECK_MODE_NORMAL, $enable = null)
```

Comment:
> Do you really think people are going to be busily setting and unsetting specific flags over the course of a program?

Not so much that - I had two specific use-cases in mind:
  1. If a flag is default-enabled internally, and the user doesn't know that, but wants to set a flag of their own - `addConfig`() is clearly safer in that situation.
  2. If a user wishes to disable a single default-enabled flag without messing up whatever else might be in `$checkMode`.

At the moment the default is just `CHECK_MODE_NORMAL`, but that may not always be the case.